Terminal.Gui/Views/Button.cs

```diff
@@ -62,13 +68,35 @@ public Button ()
         CanFocus = true;
 
         // Override default behavior of View
-        AddCommand (Command.HotKey, () =>
-        {
-            SetFocus ();
-            return !OnAccept ();
-        });
+        AddCommand (
+                    Command.HotKey,
+                    () =>
+                    {
+                        bool cachedIsDefault = IsDefault; // Supports "Swap Default" in Buttons scenario where IsDefault changes
```

Comment:
This is quite big for a lamda, would consider putting in a named method

Reply:
Good call.